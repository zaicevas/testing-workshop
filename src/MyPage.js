import { useState } from "react";

const MyPage = ({ title }) => {
  const [mutableTitle, setMutableTitle] = useState(title);

  const handleChangeTitleClick = () => {
    setMutableTitle("Mutated title");
  };

  return (
    <>
      {mutableTitle}
      <button onClick={handleChangeTitleClick}>Change title</button>
    </>
  );
};

export default MyPage;
