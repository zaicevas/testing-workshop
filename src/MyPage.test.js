import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import MyPage from "./MyPage";

describe("<MyPage />", () => {
  it("renders correctly", () => {
    const { container } = render(<MyPage title="Hello world" />);

    expect(container).toMatchSnapshot();
  });

  it("updates title after click", () => {
    render(<MyPage title="Hello world" />);

    userEvent.click(screen.getByText("Change title"));

    expect(screen.getByText("Mutated title")).toBeInTheDocument();
  });
});
