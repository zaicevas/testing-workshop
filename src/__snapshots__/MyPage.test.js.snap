// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyPage /> renders correctly 1`] = `
<div>
  Hello world
  <button>
    Change title
  </button>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  Hello world
  <button>
    Change title
  </button>
</div>
`;
